projects/packages/changelogger/src/Plugins/WordpressVersioning.php

```diff
@@ -182,18 +192,28 @@ public function compareVersions( $a, $b ) {
 		$bb = $this->parseVersion( $b );
 		if ( $aa['major'] !== $bb['major'] ) {
 			return $aa['major'] < $bb['major'] ? -1 : 1;
-
 		}
 		if ( $aa['point'] !== $bb['point'] ) {
 			return $aa['point'] - $bb['point'];
 		}
 
-		list( $aindex, $acount ) = $this->parsePrerelease( $aa['prerelease'] );
-		list( $bindex, $bcount ) = $this->parsePrerelease( $bb['prerelease'] );
+		list( $aindex, $avalues ) = $this->parsePrerelease( $aa['prerelease'] );
+		list( $bindex, $bvalues ) = $this->parsePrerelease( $bb['prerelease'] );
+
 		if ( $aindex !== $bindex ) {
 			return $aindex - $bindex;
 		}
-		return $acount - $bcount;
+
+		// Prerelease version comparison on specific pattern 'x.y-YYMM.z' such as '10.2-2112.2'.
+		if ( 2 === $aindex && 2 === $bindex ) {
+			if ( $avalues[0] !== $bvalues[0] ) {
+				return $avalues[0] - $bvalues[0];
+			} elseif ( $avalues[1] !== $bvalues[1] ) {
+				return $avalues[1] - $bvalues[1];
+			}
+		}
+
+		return $avalues[0] - $bvalues[0];
```

Comment:
More straightforward to do like this
```suggestion
		$l = min( count( $avalues ), count( $bvalues ) );
		for ( $i = 0; $i < $l; $i++ ) {
			if ( $avalues[ $i ] !== $bvalues[ $i ] ) {
				return $avalues[ $i ] - $bvalues[ $i ];
			}
		}
		return count( $avalues ) - count( $bvalues );
```

Even more straightforward would be to flatten the array returned from `parsePrerelease` and drop lines 203–205 too. The flattening might wind up as something like
```php
	if ( preg_match( "/^{$re}\$/", $s, $m ) ) {
		$m[0] = $i;
		return array_map( 'intval', $m );
	}
```

Reply:
@anomiex Awesome! I've added this in https://github.com/Automattic/jetpack/pull/21140/commits/0968a2e010a54d74abe0cb7481e7b1fc1091ae2f

And just to confirm, `'1.1.1-alpha', '==', '1.1.1-alpha0'` ?

Reply:
> And just to confirm, `'1.1.1-alpha', '==', '1.1.1-alpha0'` ?

Yes. Someone might also make a case for '<', but '==' has been the historical behavior. And it's not terribly likely someone will actually use both "alpha" and "alpha0" anyway.